### DISCUSSION TITLE
useWatch dependency chains?

### DISCUSSION THREAD

**Discussion by mphash** [react-hook-form/react-hook-form#3514] [Q&A]
Hello,

I am translating a large (~1000 input fields) excel-based spreadsheet into a web form. Here, there are several fields that are calculated based on input fields and whose results are used for other fields. Since the form is very large, I must avoid full rerenders if at all possible. I tried using useWatch for calculations, like in this component here:

```javascript
function TriControlRow(props) {
  const classes = useStyles();
  const watchid = props.watchid || props.overid;
  const tw = useWatch({name: watchid, defaultValue:"0", control:props.control});

  return (
    <tr key={"row_"+props.finid}>
      {props.label && <td>{props.label}</td>}
      <td><FormControl className={classes.smallTightControl}><TextField id={props.calc} name={props.calc} disabled/></FormControl></td>
      <td>
        <FormControl className={classes.smallTightControl}>
            <TextField id={props.overid} name={props.overid}
                   className={classes.overrideField}
                   inputRef={props.register} />
        </FormControl>
      </td>
      <td>
        <FormControl className={classes.smallTightControl}>
          <TextField inputRef={props.register}
              id={props.finid} value={tw || "0"} control = {props.control} name={props.finid} disabled />
        </FormControl>
      </td>
    </tr>
  );
}
```

This works when I try to watch override (overid) fields, but unfortunately, when I try to watch another component's result (watchid of one equal to finid of other), it fails to update, even though it works when watchid is equal to the overid of the active input field. 

This does not work:
```javascript
      <TriControlRow label="foo"  calc="F53" overid={"foover"} finid={"foofin"}  control={control} register={register} />
      <TriControlRow label="bar" calc="F59" overid={"barover"} finid={"barin"} watchid={"foofin"} control={control} register={register}/>

```

Is it possible to propagate updates this way, or am I missing something?

EDIT: [Code sandbox link here](https://codesandbox.io/s/bitter-framework-4frxs?file=/index.js)

**Comment by bluebill1049** [comment]
can you share your usage in a codesandbox?

**Reply by mphash** [reply]
Sure. Sorry for the long wait; we seem to be in different timezones.

[Code sandbox link here](https://codesandbox.io/s/bitter-framework-4frxs?file=/index.js)

**Reply by bluebill1049** [reply]
what's the steps to reproduce the problem at your CSB above?

**Reply by mphash** [reply]
The last field of "baz" should update when you type something in the second (orange) field of bar, but it does not.
